### ISSUE TITLE
Offline Plugin not working with nginx configuration

### ISSUE THREAD

**Issue by talbenbasat** [react-boilerplate/react-boilerplate#1687]
I'm using react-boilerplate v3.2.1 and deployed my system with nginx and django backend.

Lately i've found out the offline plugin isn't working with my nginx configuration.
I've tried using parts of the configurations in:
https://github.com/react-boilerplate/react-boilerplate/blob/master/app/.nginx.conf
But it still not working.

When I run the project with  `npm run start:production` it works.

this is my nginx configuration:

```
upstream app_server {
  # Bindings to the Gunicorn server
  server 127.0.0.1:8002 fail_timeout=0;
}

server {

  # Access Logs
  access_log /home/ubuntu/webapps/myproject/logs/nginx-access.log;
  error_log /home/ubuntu/webapps/myproject/logs/nginx-error.log;

  listen 80;
  client_max_body_size 4G;

  # set server name - server_name _;
  server_name myproject.net;

  return 301 $scheme://www.myproject.net$request_uri;
}



server {
  
  # Access Logs
  access_log /home/ubuntu/webapps/myproject/logs/nginx-access.log;
  error_log /home/ubuntu/webapps/myproject/logs/nginx-error.log;


  listen 80 default;
  client_max_body_size 4G;

  # set server name - server_name _;
  server_name www.myproject.net;

  keepalive_timeout 5;
  root   /var/www/myproject/version118;

  location /admin {
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header Host $http_host;
    proxy_redirect off;
    proxy_pass http://app_server/admin;
  }

  location /api {
    add_header Cache-Control "no-cache, must-revalidate, max-age=0";

    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header Host $http_host;
    proxy_set_header Origin $http_origin;
    proxy_set_header Agent $http_agent;

    proxy_redirect off;
    proxy_pass http://app_server/api;
  }

  location /chaining {
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header Host $http_host;
    proxy_redirect off;
    proxy_pass http://app_server/chaining;
  }

  location /social {
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header Host $http_host;
    proxy_redirect off;
    proxy_pass http://app_server/social;
  }

  location /sitemap/ {
    alias   /var/www/sitemap/;
  }

  location /sitemap.xml {
    root   /var/www/sitemap/;
  }

  location /robots.txt {
    root   /var/www/robots/;
  }

  location /static/ {
    alias   /var/www/static/;
  }

  location ~ /default.aspx {
    if ($args ~* "^pageid=7&stoneId=(\d+)") {
      set $mid $1;
      return 301 /memorials/$mid;
    }
    return 301 /;
  }

  location ~ ^/$ {
    if ($args ~* "^pageid=7&stoneId=(\d+)") {
      set $mid $1;
      return 301 /memorials/$mid;
    }

    try_files $uri  $uri @prerender;
  }


  location / {
    if ($request_uri ~* ".(jpg|jpeg|gif|gz|zip|flv|rar|wmv|avi|css|js|swf|png|htc|ico|mpeg|mpg|txt|mp3|mov|woff)(\?v=[0-9.]+)?$") {
      expires 30d;
      access_log off;
      break;
    }

    try_files $uri @prerender;
  }


  location @prerender {
    error_page 404 =200 /index.html;

    set $prerender 0;
    if ($http_user_agent ~* "googlebot|baiduspider|twitterbot|facebookexternalhit|rogerbot|linkedinbot|embedly|quora link preview|showyoubot|outbrain|pinterest|slackbot|vkShare|W3C_Validator") {
        set $prerender 1;
    }
    if ($args ~ "_escaped_fragment_") {
        set $prerender 1;
    }
    if ($http_user_agent ~ "Prerender") {
        set $prerender 0;
    }
    if ($uri ~ "\.(js|css|xml|less|png|jpg|jpeg|gif|pdf|doc|txt|ico|rss|zip|mp3|rar|exe|wmv|doc|avi|ppt|mpg|mpeg|tif|wav|mov|psd|ai|xls|mp4|m4a|swf|dat|dmg|iso|flv|m4v|torrent|ttf|woff)") {
        set $prerender 0;
    }

    # resolve using Googles DNS server to force DNS resolution and prevent caching of IPs
    resolver 8.8.8.8;

    if ($prerender = 1) {
        # prerender is running locally in port 3000
        proxy_pass http://127.0.0.1:3000/http://www.myproject.net$request_uri;
    }
  }

  location /sw.js {
      add_header Cache-Control "no-cache";
      proxy_cache_bypass $http_pragma;
      # proxy_cache_revalidate on;
      expires off;
      access_log off;
  }

  error_page 500 502 503 504 /500.html;
  location = /500.html {
    root www.myproject.net;
  }
}
```

Any idea how It can be solved?

**Comment by thomas88** [comment]
I'm using the nginx template as well and had to nest the `/sw.js` location under the `/` location, so that it knows where to look for the file.

**Comment by carlos4ndre** [comment]
I had the same issue and ended up adding an `alias` attribute to access the `sw.js` file.

```
# Do not cache sw.js, required for offline-first updates.
location /sw.js {
  alias /var/www/website/sw.js;
  add_header Cache-Control "no-cache";
  proxy_cache_bypass $http_pragma;
  proxy_cache_revalidate on;
  expires off;
  access_log off;
}
```

**Comment by blling** [comment]
Fixed in #2047, if you have some individual locations you can try what @carlos4ndre said.
